### ISSUE TITLE
test_spectral_irf.py Aborted due to Fatal Python error

### ISSUE THREAD

**Issue by jsnel** [glotaran/pyglotaran#352]
* glotaran version:
* Python version: 3.6, 3.7, 3.8
* Operating System: Linux 5.6.14-1-default 

### Description

test_spectral_irf more often than not crashes with a core dump
sometimes when the system is under load, or the test is run in debug mode, it does complete 

### What I Did
The the command:
`pytest glotaran/builtin/models/kinetic_spectrum/test/test_spectral_irf.py`

```
=== test session starts ===
platform linux -- Python 3.8.2, pytest-5.4.2, py-1.8.1, pluggy-0.13.1
benchmark: 3.2.3 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=False warmup_iterations=100000)
rootdir: pyglotaran
plugins: cov-2.9.0, benchmark-3.2.3
collected 2 items                                                                                                                                                                                                                                                             

glotaran/builtin/models/kinetic_spectrum/test/test_spectral_irf.py .Fatal Python error: Aborted

Thread 0x00007f5dabfff700 (most recent call first):
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/multiprocessing/pool.py", line 576 in _handle_results
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/threading.py", line 870 in run
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/threading.py", line 932 in _bootstrap_inner
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/threading.py", line 890 in _bootstrap

Thread 0x00007f5db0ff9700 (most recent call first):
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/multiprocessing/pool.py", line 528 in _handle_tasks
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/threading.py", line 870 in run
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/threading.py", line 932 in _bootstrap_inner
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/threading.py", line 890 in _bootstrap

Thread 0x00007f5db17fa700 (most recent call first):
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/selectors.py", line 415 in select
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/multiprocessing/connection.py", line 930 in wait
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/multiprocessing/pool.py", line 499 in _wait_for_updates
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/multiprocessing/pool.py", line 519 in _handle_workers
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/threading.py", line 870 in run
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/threading.py", line 932 in _bootstrap_inner
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/threading.py", line 890 in _bootstrap

Thread 0x00007f5db1ffb700 (most recent call first):
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/multiprocessing/pool.py", line 114 in worker
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/threading.py", line 870 in run
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/threading.py", line 932 in _bootstrap_inner
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/threading.py", line 890 in _bootstrap

Thread 0x00007f5db27fc700 (most recent call first):
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/multiprocessing/pool.py", line 114 in worker
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/threading.py", line 870 in run
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/threading.py", line 932 in _bootstrap_inner
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/threading.py", line 890 in _bootstrap

Thread 0x00007f5db2ffd700 (most recent call first):
  File "pyglotaran/glotaran/builtin/models/kinetic_image/kinetic_image_matrix.py", line 111 in kinetic_image_matrix_implementation
  File "pyglotaran/glotaran/builtin/models/kinetic_image/kinetic_image_matrix.py", line 89 in _calculate_for_k_matrix
  File "pyglotaran/glotaran/builtin/models/kinetic_image/kinetic_image_matrix.py", line 37 in kinetic_matrix
  File "pyglotaran/glotaran/builtin/models/kinetic_image/kinetic_image_matrix.py", line 13 in kinetic_image_matrix
  File "pyglotaran/glotaran/builtin/models/kinetic_spectrum/kinetic_spectrum_matrix.py", line 13 in kinetic_spectrum_matrix
  File "pyglotaran/glotaran/analysis/matrix_calculation.py", line 137 in _calculate_matrix
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/dask/utils.py", line 30 in apply
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/dask/core.py", line 121 in _execute_task
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/dask/local.py", line 222 in execute_task
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/multiprocessing/pool.py", line 125 in worker
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/threading.py", line 870 in run
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/threading.py", line 932 in _bootstrap_inner
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/threading.py", line 890 in _bootstrap

Thread 0x00007f5db37fe700 (most recent call first):
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/multiprocessing/pool.py", line 114 in worker
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/threading.py", line 870 in run
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/threading.py", line 932 in _bootstrap_inner
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/threading.py", line 890 in _bootstrap

Thread 0x00007f5db3fff700 (most recent call first):
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/multiprocessing/pool.py", line 114 in worker
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/threading.py", line 870 in run
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/threading.py", line 932 in _bootstrap_inner
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/threading.py", line 890 in _bootstrap

Current thread 0x00007f5db891e700 (most recent call first):
  File "pyglotaran/glotaran/builtin/models/kinetic_image/kinetic_image_matrix.py", line 111 in kinetic_image_matrix_implementation
  File "pyglotaran/glotaran/builtin/models/kinetic_image/kinetic_image_matrix.py", line 89 in _calculate_for_k_matrix
  File "pyglotaran/glotaran/builtin/models/kinetic_image/kinetic_image_matrix.py", line 37 in kinetic_matrix
  File "pyglotaran/glotaran/builtin/models/kinetic_image/kinetic_image_matrix.py", line 13 in kinetic_image_matrix
  File "pyglotaran/glotaran/builtin/models/kinetic_spectrum/kinetic_spectrum_matrix.py", line 13 in kinetic_spectrum_matrix
  File "pyglotaran/glotaran/analysis/matrix_calculation.py", line 137 in _calculate_matrix
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/dask/utils.py", line 30 in apply
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/dask/core.py", line 121 in _execute_task
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/dask/local.py", line 222 in execute_task
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/multiprocessing/pool.py", line 125 in worker
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/threading.py", line 870 in run
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/threading.py", line 932 in _bootstrap_inner
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/threading.py", line 890 in _bootstrap

Thread 0x00007f5db911f700 (most recent call first):
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/multiprocessing/pool.py", line 114 in worker
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/threading.py", line 870 in run
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/threading.py", line 932 in _bootstrap_inner
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/threading.py", line 890 in _bootstrap

Thread 0x00007f5db9920700 (most recent call first):
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/multiprocessing/pool.py", line 114 in worker
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/threading.py", line 870 in run
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/threading.py", line 932 in _bootstrap_inner
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/threading.py", line 890 in _bootstrap

Thread 0x00007f5dede56740 (most recent call first):
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/threading.py", line 302 in wait
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/queue.py", line 170 in get
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/dask/local.py", line 133 in queue_get
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/dask/local.py", line 475 in get_async
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/dask/threaded.py", line 76 in get
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/dask/base.py", line 444 in compute
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/dask/base.py", line 166 in compute
  File "pyglotaran/glotaran/analysis/optimize.py", line 107 in calculate_penalty
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/lmfit/minimizer.py", line 576 in __residual
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/scipy/optimize/_lsq/least_squares.py", line 802 in fun_wrapped
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/scipy/optimize/_lsq/least_squares.py", line 807 in least_squares
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/lmfit/minimizer.py", line 1527 in least_squares
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/lmfit/minimizer.py", line 2285 in minimize
  File "pyglotaran/glotaran/analysis/optimize.py", line 54 in optimize_task
  File "pyglotaran/glotaran/analysis/optimize.py", line 34 in optimize
  File "pyglotaran/glotaran/model/model.py", line 185 in optimize
  File "pyglotaran/glotaran/builtin/models/kinetic_spectrum/test/test_spectral_irf.py", line 254 in test_spectral_irf
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/_pytest/python.py", line 182 in pytest_pyfunc_call
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/pluggy/callers.py", line 187 in _multicall
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/pluggy/manager.py", line 84 in <lambda>
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/pluggy/hooks.py", line 286 in __call__
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/_pytest/python.py", line 1477 in runtest
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/_pytest/runner.py", line 135 in pytest_runtest_call
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/pluggy/callers.py", line 187 in _multicall
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/pluggy/manager.py", line 84 in <lambda>
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/pluggy/hooks.py", line 286 in __call__
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/_pytest/runner.py", line 217 in <lambda>
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/_pytest/runner.py", line 244 in from_call
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/_pytest/runner.py", line 216 in call_runtest_hook
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/_pytest/runner.py", line 186 in call_and_report
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/_pytest/runner.py", line 100 in runtestprotocol
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/_pytest/runner.py", line 85 in pytest_runtest_protocol
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/pluggy/callers.py", line 187 in _multicall
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/pluggy/manager.py", line 84 in <lambda>
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/pluggy/hooks.py", line 286 in __call__
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/_pytest/main.py", line 272 in pytest_runtestloop
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/pluggy/callers.py", line 187 in _multicall
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/pluggy/manager.py", line 84 in <lambda>
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/pluggy/hooks.py", line 286 in __call__
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/_pytest/main.py", line 247 in _main
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/_pytest/main.py", line 191 in wrap_session
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/_pytest/main.py", line 240 in pytest_cmdline_main
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/pluggy/callers.py", line 187 in _multicall
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/pluggy/manager.py", line 84 in <lambda>
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/pluggy/hooks.py", line 286 in __call__
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/lib/python3.8/site-packages/_pytest/config/__init__.py", line 124 in main
  File "~/anaconda3/envs/pyglotaran_38_dev_pip/bin/pytest", line 8 in <module>
Aborted (core dumped)


```

**Comment by joernweissenborn** [comment]
@jsnel Does this issue still persists?

**Comment by jsnel** [comment]
This was solved prior to v0.3.0 release (possibly already when switching to our internal optimization framework).
